internal/dag/cache.go

```diff
@@ -179,39 +191,47 @@ func (kc *KubernetesCache) Insert(obj interface{}) bool {
 					kc.gatewayclass = gatewayClass
 				}
 
-				return true
+				return true, 1
 			// Otherwise, take whatever we're given.
 			default:
 				kc.gateway = obj
-				return true
+				return true, 1
 			}
+
 		case *gatewayapi_v1beta1.HTTPRoute:
 			kc.httproutes[k8s.NamespacedNameOf(obj)] = obj
-			return kc.routeTriggersRebuild(obj.Spec.ParentRefs)
+			return kc.routeTriggersRebuild(obj.Spec.ParentRefs), len(kc.httproutes)
+
 		case *gatewayapi_v1alpha2.TLSRoute:
 			kc.tlsroutes[k8s.NamespacedNameOf(obj)] = obj
-			return kc.routeTriggersRebuild(obj.Spec.ParentRefs)
+			return kc.routeTriggersRebuild(obj.Spec.ParentRefs), len(kc.tlsroutes)
+
 		case *gatewayapi_v1beta1.ReferenceGrant:
 			kc.referencegrants[k8s.NamespacedNameOf(obj)] = obj
-			return true
+			return true, len(kc.referencegrants)
+
 		case *contour_api_v1alpha1.ExtensionService:
 			kc.extensions[k8s.NamespacedNameOf(obj)] = obj
-			return true
+			return true, len(kc.extensions)
+
 		case *contour_api_v1alpha1.ContourConfiguration:
-			return false
+			return false, 0
+
 		default:
 			// not an interesting object
 			kc.WithField("object", obj).Error("insert unknown object")
-			return false
+			return false, 0
 		}
 	}
 
-	if maybeInsert(obj) {
+	if ok, count := maybeInsert(obj); ok {
 		// Only check annotations if we actually inserted
 		// the object in our cache; uninteresting objects
 		// should not be checked.
 		if obj, ok := obj.(metav1.Object); ok {
 			kind := k8s.KindOf(obj)
```

Comment:
Please double-check that `KindOf` actually covers all the resource types we cache -- IIRC it was missing some (ref. #5012)

Reply:
yes, the `KindOf` not covers the all resource types now, but I don't know whether I should update it or leave it to #5012

Reply:
OK let's get #5012 merged and then we'll be good here.